src/accessibility/describe.js

```diff
@@ -235,131 +170,275 @@ p5.prototype.describeElement = function(name, text, display) {
     return;
   }
   const cnvId = this.canvas.id;
+  //calls function that adds punctuation for better screen reading
   text = _descriptionText(text);
+  //calls function that adds punctuation for better screen reading
   let elementName = _elementName(name);
-  name = _nameForID(name);
+  //remove any special characters from name to use it as html id
+  name = name.replace(/[^a-zA-Z0-9 ]/g, '');
+  //store element description
   let inner = `<th scope="row">${elementName}</th><td>${text}</td>`;
 
+  //if it is the first time describeElement() is called
   if (
     !dummy.fallbackElements[cnvId + name] ||
     !dummy.labelElements[cnvId + name]
   ) {
+    //store copy of body dom in dummy
     _populateDummyDOM(cnvId);
   }
 
+  //check if element description is different from current
   if (dummy.fallbackElements[cnvId + name] !== inner) {
-    if (!dummy.fallbackElements[cnvId + name]) {
-      _descElementFallbackHTML(cnvId, name, inner);
-    } else {
-      dummy.fallbackElements[cnvId + name] = inner;
+    //if html structure for element description is ready
+    if (dummy.fallbackElements[cnvId + name]) {
+      //update element description
       dummy[cnvId + 'DOM'].querySelector(
         '#' + cnvId + fallbackTableElId + name
       ).innerHTML = inner;
+      //store updated element description
+      dummy.fallbackElements[cnvId + name] = inner;
+    } else {
+      //create fallback html structure
+      _descElementFallbackHTML(cnvId, name, inner);
     }
   }
-  //If display is LABEL creates a div adjacent to the canvas element with
-  //a table, a row header cell with the name of the elements,
-  //and adds the description of the element in adjecent cell.
+  //if display is LABEL and label element description is different
   if (display === this.LABEL && dummy.labelElements[cnvId + name] !== inner) {
-    if (!dummy.labelElements[cnvId + name]) {
-      _descElementLabelHTML(cnvId, name, inner);
-    } else {
-      dummy.labelElements[cnvId + name] = inner;
+    //if html structure for label element description is ready
+    if (dummy.labelElements[cnvId + name]) {
+      //update label element description
       dummy[cnvId + 'DOM'].querySelector(
         '#' + cnvId + labelTableElId + name
       ).innerHTML = inner;
+      //store updated label element description
+      dummy.labelElements[cnvId + name] = inner;
+    } else {
+      //create label element html structure
+      _descElementLabelHTML(cnvId, name, inner);
     }
   }
 };
-/**
- * Helper functions for describeElement().
+
+/*
+ *
+ * Helper functions for describe() and describeElement().
+ *
  */
-function _descElementLabelHTML(cnvId, name, inner) {
-  if (!dummy[cnvId + labelContainer]) {
+
+//clear dummy
+p5.prototype._clearDummy = function() {
```

Comment:
I'm curious why this method is attached to the prototype but not the fallback elements or label elements. Wondering if it makes sense for fallbackElements and labelElements to be attached to a p5 instance? (I recall that this got added to fix the tests, am wondering if attaching to the instance would also fix that or if there are other possible advantages)